### PR TITLE
Rust: add mentoring notes for gigasecond

### PR DESCRIPTION
**Pull request by TheDarkula**
The unique thing about this exercise is that we can teach students about the concept of constants in a very approachable way. It's a small performance boost that will teach the best-practise of "CONST ALL THE THIIIIINGS!" :)

### CODE REVIEW

File: tracks/rust/exercises/gigasecond/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+- Use constants
+- Use underscores in place of commas
+- Be explicit with use lines
+- Not use `return`
+- Not use methods like `.add()` or `.checked_add_signed()`
+- Not use variables
+- Not use `.unwrap()` or `.expect()`
+- Not use `.pow()`
+
+### Examples
```
Comment (SleeplessByte) [inline tracks/rust/exercises/gigasecond/mentoring.md]: Change to level 4

File: tracks/rust/exercises/gigasecond/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+use chrono::{DateTime, Duration, Utc};
+
+const GIGASECOND: i64 = 1_000_000_000;
+
+pub fn after(start: DateTime<Utc>) -> DateTime<Utc> {
+    start + Duration::seconds(GIGASECOND)
+}
+```
+
+### Example Comments
```
Comment (SleeplessByte) [inline tracks/rust/exercises/gigasecond/mentoring.md]: Change to 
`### Common Suggestions`

File: tracks/rust/exercises/gigasecond/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+### Concepts
+
+- Integer
+- constant
+- struct (We're returning the `Utc` struct, which is the field in the
```
Comment (SleeplessByte) [inline tracks/rust/exercises/gigasecond/mentoring.md]: You might want to consider linking `struct`, `Utc` and `DateTime` to the official docs/reference and `Chrono crate` to the chrono crate page.

File: tracks/rust/exercises/gigasecond/mentoring.md (hunk 1 above)
Comment (TheDarkula) [reply tracks/rust/exercises/gigasecond/mentoring.md]: Done :)

File: tracks/rust/exercises/gigasecond/mentoring.md (hunk 2 above)
Comment (TheDarkula) [reply tracks/rust/exercises/gigasecond/mentoring.md]: Done :)

File: tracks/rust/exercises/gigasecond/mentoring.md (hunk 3 above)
Comment (TheDarkula) [reply tracks/rust/exercises/gigasecond/mentoring.md]: Great thought! Done :)

**Comment by TheDarkula** [comment]
@SleeplessByte Thanks so much for the helpful review!

**Comment by TheDarkula** [comment]
@SleeplessByte I'm adding the standard library links now :)

**Comment by TheDarkula** [comment]
@SleeplessByte All done :) Thanks again!

**Comment by SleeplessByte** [comment]
Perfect @TheDarkula! 

As with the other one, do you perhaps want to wait on a review by @coriolinus? I know he's the other one on @exercism/website-copy-rust. If not, I'm okay with merging this instead of waiting and making subsequent changes on a new PR, as having some notes are better than no notes.

**Comment by TheDarkula** [comment]
@SleeplessByte I agree with merging this now since it is a core exercise and we have no notes currently. Thanks again for the help and approving this :)

@coriolinus If there are any improvements you can think of, let me know in the #track-rust channel.

Merged :)

**Pull request merged**